### PR TITLE
@cbrown/non employee rbac

### PR DESCRIPTION
**Pull request by Collinbrown95**
Refactor `deny-external-users` and `deny-employee-only-features` policies to adhere to new labelling scheme

### CODE REVIEW

File: general/deny-external-users/rego/src.rego [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package denyexternalusers
+
+# RoleBinding Object
+violation[{"msg": msg}] {
+    obj := input.review.object
```
Comment (Souheil-Yazji) [inline general/deny-external-users/rego/src.rego:5]: I may be mistaken but all of these re-occurring assignments should be extractable out of block, leaving only eval logic in violation blocks.

**Review by Souheil-Yazji — APPROVED** [review]
LGTM with comment

**Review by chuckbelisle — APPROVED** [review]
lgtm

File: general/deny-external-users/rego/src.rego (hunk 1 above)
Comment (Collinbrown95) [reply general/deny-external-users/rego/src.rego:5]: Valid point, but I will refactor this in a separate PR

**Comment by sylus** [comment]
The policy and package are called employee only but it looks like it's now tailored for sas exclusive logic.

I am thinking the overly broad name doesn't align to its implementation anymore?

**Comment by justbert** [comment]
> The policy and package are called employee only but it looks like it's now tailored for sas exclusive logic.
> 
> I am thinking the overly broad name doesn't align to its implementation anymore?

Agreed here. It might be useful to start splitting these into specific AAW folders.

**Comment by Collinbrown95** [comment]
@sylus @justbert I am good with this proposed change. Would you like me to move `deny-external-users` and `deny-employee-only-features` to a new top-level `aaw` folder? I can also change the folder/policy names to more accurately reflect the semantics of the policies.

File: general/deny-external-users/rego/src.rego [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    email := obj.spec.rules[_]["when"][_]["values"][_]
+    contains(email, "@")
+    not isEmployee(email)
+    # If the non-employee is in the exception list, then allow
+    count([exceptionCase | exceptionCase := input.parameters.sasNotebookExceptions[_]; exceptionCase == email]) == 0
+    ns := obj.metadata.namespace
+    profile := data.inventory.cluster["kubeflow.org/v1"]["Profile"][ns]
+    profileLabel := input.parameters.labels[_]
+    profile.metadata.labels[profileLabel] == "true"
+    msg := sprintf("Profile %v has %v=%v", [ns, profileLabel, profile.metadata.labels[profileLabel]])
```
Comment (justbert) [inline general/deny-external-users/rego/src.rego:32]: You might think about moving this into a function. 😄

**Review by justbert — COMMENTED** [review]
Generally looks good. A few suggestions :)

**Comment by Collinbrown95** [comment]
@justbert for the changes discussed in https://github.com/StatCan/gatekeeper-policies/pull/46#issuecomment-1252404446, do you want me to perform this renaming + moving of these policies in this PR, or should this be left to a separate PR to create the AAW folder you and @sylus mentioned?

File: pod-security-policy/deny-employee-only-features/rego/src.rego [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    existsSasContainerAndLabelViolation(container, profileLabel)
+    msg := sprintf("Pod has state.aaw.statcan.gc.ca/exists-non-sas-notebook-user=%v and container uses a SAS image %v", [profileLabel, container.image])
+}
+
+# Notebook Object
+violation[{"msg": msg}] {
+    input.review.object.kind == "Notebook"
+    container := input.review.object.spec.template.spec.containers[_]
+    existsSasContainerAndLabelViolation(container, profileLabel)
+    msg := sprintf("Notebook has state.aaw.statcan.gc.ca/exists-non-sas-notebook-user=%v and container uses a SAS image %v", [profileLabel, container.image])
```
Comment (justbert) [inline pod-security-policy/deny-employee-only-features/rego/src.rego:23]:
```suggestion
    msg := sprintf("Notebook has %s=%v and container uses a SAS image %v", [sasLabel, profileLabel, container.image])
```

**Review by justbert — COMMENTED** [review]
Just a few more suggestions! They are not requirements, I'm just hoping to help make you think and help make the code clearer. :)

I hope you find the same, but if you disagree, that's totally okay!

**Review by justbert — APPROVED** [review]
In general, looks good to me! Hope you don't mind all my comments. As for the new code structure, I'll differ to @sylus.

**Comment by Collinbrown95** [comment]
@justbert  no problem at all, I appreciate the feedback :)

Sounds good re: new code structure; @sylus please let me know which you prefer and I'll either do the renaming + folder reorganization as part of this PR or create a separate PR shortly with the recommended reorganization.

**Comment by sylus** [comment]
Hi @Collinbrown95 its fine to do the rest as a subsequent P.R. so can review just that change :)

Thanks!

**Comment by Collinbrown95** [comment]
That sounds good to me, thanks @sylus !

**Pull request merged**